src/control/security/auth/auth_sys.go

```diff
@@ -46,43 +46,63 @@ type UserExt interface {
 	LookupGroupID(gid uint32) (*user.Group, error)
 }
 
-// HashFromToken will return a SHA512 hash of the token data
-func HashFromToken(token *auth.Token) ([]byte, error) {
-	// Generate our hash (not signed yet just a hash)
-	hash := sha512.New()
+//VerifierFromToken will return a SHA512 hash of the token data. If a signing key
```

Comment:
This comment seems to be wrong now, unless I'm misunderstanding the code.

Reply:
How is it wrong?

Reply:
Ah, I did misunderstand the code. signer.Sign() can take a nil key, I guess.